src/date.test.ts

```diff
@@ -118,3 +119,28 @@ describe('formatGermanFullDateTime', () => {
     );
   });
 });
+
+describe('dateMatchesFormat', () => {
+  it('recognizes valid dates', () => {
+    expect(dateMatchesFormat('20240101', 'yyyyMMdd')).toBe(true);
```

Comment:
Is the format "yyyyMMdd" necessary or is "Ymd" also possible?

Reply:
Produces the following error:

```
The format string mustn't contain `Y` and `d` at the same time
RangeError: The format string mustn't contain `Y` and `d` at the same time
    at parse (/home/bfranke/projects/js-utils/node_modules/date-fns/parse.js:415:17)
    at isMatch (/home/bfranke/projects/js-utils/node_modules/date-fns/isMatch.js:298:23)
    at dateMatchesFormat (/home/bfranke/projects/js-utils/src/date.ts:111:17)
```